### PR TITLE
added grid iterator and access through cartesian indices

### PR DESCRIPTION
**Pull request by mattuntergassmair**
implements iteration functionality from issue #14

### CODE REVIEW

File: src/GridInterpolations.jl [hunk 1]
```diff
@@ -369,4 +369,10 @@ function sortperm!(x::Vector{I}, v::AbstractVector; alg::Algorithm=DEFAULT_UNSTA
     sort!(x, alg, Perm(ord(lt,by,rev,order),v))
 end
 
+function Base.iterate(iter::RectangleGrid, state::Int64=1)
+    return state<=length(iter) ? (ind2x(iter, state), state+1) : nothing
+end
+
+Base.getindex(grid::RectangleGrid, key::CartesianIndex) = ind2x(grid, LinearIndices(Dims((grid.cut_counts...,)))[key])
```
Comment (MaximeBouton) [inline src/GridInterpolations.jl]: Can you document this in the README?

File: test/runtests.jl [hunk 2]
```diff
@@ -358,6 +358,31 @@ end
...
 end
 
+@testset "Iteration and Indexing" begin
+    @testset "2d" begin
+        g = GridInterpolations.RectangleGrid(1:3,4:5)
+        pts = ([1.,4.],[2.,4.],[3.,4.],[1.,5.],[2.,5.],[3.,5.])
+        @test length(g) == length(pts)
+        @test all(pts .== [x for x in g])
+        @test iterate(g,7) === nothing
+        @test all(pts .== [g[ci] for ci in reshape(CartesianIndices((3,2)), length(g))])
```
Comment (MaximeBouton) [inline test/runtests.jl]: Is the `CartesianIndices` constructor needed or can one just index using `g[3, 2]` for example?

Comment (mattuntergassmair) [reply test/runtests.jl]: Support for multi-indices added in latest commit (although Cartesian coordinates seem to be safer generally)

**Comment by MaximeBouton** [comment]
Tests are failing on julia 0.7: https://travis-ci.org/sisl/GridInterpolations.jl/jobs/594904026
It seems to be a bounds error issue. If you see a fix go for it otherwise we can drop the support for 0.7. 

On Travis we could add osx, windows and julia 1.2.

**Comment by mattuntergassmair** [comment]
@MaximeBouton I am not sure where the error is coming from exactly but my best guess is that it's a change of implementation of `LinearIndices` itself. The failing test uses two indices to index a 3d grid and expects a `BoundsError`, but it seems this is not thrown in julia <=v0.7

**Comment by mattuntergassmair** [comment]
Just added a condition to skip the test for v < 1.0

**Comment by rejuvyesh** [comment]
I think we can safely drop support for 0.7 now.

**Pull request merged**